frag-nav/src/main/java/com/ncapdevi/fragnav/FragNavTransactionOptions.java

```diff
@@ -0,0 +1,78 @@
+package com.ncapdevi.fragnav;
+
+import android.support.annotation.AnimRes;
+import android.support.annotation.StyleRes;
+import android.support.v4.util.Pair;
+import android.view.View;
+
+import java.util.List;
+
+/**
+ *
+ */
+
+
+public class FragNavTransactionOptions {
+    List<Pair<View, String>> sharedElements;
+    @FragNavController.Transit
+    Integer transition;
+    @AnimRes
+    Integer enterAnimationId;
+    @AnimRes
+    Integer exitAnimationId;
+    @StyleRes
+    Integer transitionStyle;
+
+    private FragNavTransactionOptions(Builder builder) {
+        sharedElements = builder.sharedElements;
+        transition = builder.transition;
+        enterAnimationId = builder.enterAnimationId;
+        exitAnimationId = builder.exitAnimationId;
+        transitionStyle = builder.transitionStyle;
+    }
+
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    public static final class Builder {
+        private List<Pair<View, String>> sharedElements;
+        private Integer transition;
+        private Integer enterAnimationId;
+        private Integer exitAnimationId;
+        private Integer transitionStyle;
+
+        private Builder() {
+        }
+
+        public Builder sharedElements(List<Pair<View, String>> val) {
```

Comment:
I'm actually second guessing this now. The problem that I could see arising is that, if you have only 
```
new Builder()
   .sharedElements(sharedElements)
```
Then you're always setting what the shared elements are. but if you have the

```
new Builder()
  .sharedElement(view, string)
  .sharedElement(view, string)
```

You are appending.  This means, if `sharedElements` is null, it creates it, and sets the first element, then the second, but if the `sharedElements()` is called, it overwrites that. Meaning one function adds, and one sets.  As I'm writing this, I guess that's fine, just going to add a clarifier to the start of the functinon `addSharedElement` vs `setSharedElements`